lib/solidus_content/providers/yaml.rb

```diff
@@ -2,28 +2,28 @@
 
 require 'yaml'
 
-module SolidusContent::Providers::YAML
-  def self.call(input)
-    dir = Rails.root.join(input.dig(:type_options, :path))
-    file = dir.join(input[:slug] + '.yml')
-    file = dir.join(input[:slug] + '.yaml') unless file.exist?
+module SolidusContent
+  module Providers
+    class YAML
+      class << self
+        def call(input)
+          dir = Rails.root.join(input.dig(:type_options, :path))
+          file = dir.join(input[:slug] + '.yml')
+          file = dir.join(input[:slug] + '.yaml') unless file.exist?
 
-    data = load(file)
+          data = load(file)
 
-    input.merge(data: data)
-  end
+          input.merge(data: data)
+        end
 
-  # Ruby 2.6 deprecated passing the filename as the second argument, we'll do 
-  # a feature check instead of relying on the Ruby version.
-  filename_key = YAML.method(:load).parameters.rassoc(:filename).first == :key
```

Comment:
Removing this we're losing contextualized stacktraces if there's a syntax error in the YAML, are we sure this version is preferable? Maybe it's worth mentioning this in the commit message as well.

Reply:
To avoid the feature check on the `load` method, we can use `load_file` that isn't deprecated instead.
The problem is that `load_file` doesn't support `sym_keys` by default and I called it after the file loading.